docs/dynamic-asg-overview.md

```diff
@@ -0,0 +1,107 @@
+-->
+
+**Steps taken to create and enforce dynamic ASGs**
+1. An operator creates and binds and ASG.
+1. A new job, the Policy Server ASG Syncer, polls Cloud Controller for all ASGs.
+   The syncer saves all the ASGs in the policy server DB.  This poll interval is
+   controlled by the syncer's bosh property
+   [`asg_poll_interval_seconds`](https://github.com/cloudfoundry/cf-networking-release/blob/0c5029c88e9f61bb94829b4e0b8ed6732f30f9f0/jobs/policy-server-asg-syncer/spec#L31-L33).
+1. The vxlan policy agent (VPA) polls the policy server for c2c policies _and_
+   ASG rules relavent for the apps on its cell. This poll interval is controlled
+   by the VPA's bosh property [`asg_poll_interval_seconds`](https://github.com/cloudfoundry/silk-release/blob/f1606499925ff94bc036a641d688967c7af1fef4/jobs/vxlan-policy-agent/spec#L59-L61).
+1. If the ASGs have changed, then the VPA creates the new ASGs as iptables
+   netout rules.
+
+## FAQ
+### 1. I'm not using silk. How does this change things?
+This implementation of dynamic ASGs is only for silk users.
+
+### 2. How do I enable dynamic ASGs?
+To enable dynamic ASGs the following two properties must be set:
+| job name | property name | value to enable | link to code |
+| --- | --- | --- | --- |
+| policy-server-asg-syncer | disable | false | [link](https://github.com/cloudfoundry/cf-networking-release/blob/0c5029c88e9f61bb94829b4e0b8ed6732f30f9f0/jobs/policy-server-asg-syncer/spec#L27-L29) |
+| vxlan-policy-agent | enable_asg_syncing | true | [link](https://github.com/cloudfoundry/silk-release/blob/f1606499925ff94bc036a641d688967c7af1fef4/jobs/vxlan-policy-agent/spec#L55-L57) |
+
+If you are using cf-deployment then dynamic ASGs are on by default.
+
+### 3. How do I disable dynamic ASGs?
+You can disable dynamic ASGs by using [this opsfile](https://github.com/cloudfoundry/cf-deployment/blob/604b5822259d8c889c3cdc4a2723af0e636570bb/operations/disable-dynamic-asgs.yml).
+
+### 4. I don't like waiting a few minutes for rules to be enforced. Can it be faster?
+
+You can still restart your app for immediate ASG updates.
+
+If you want to speed up the dynamic enforcement you can update the following
+bosh properties. Lowering these values will have performance implications.
+
+| job name | property name | default | link to code|
+| --- | --- | --- | --- |
+| policy-server-asg-syncer | asg_poll_interval_seconds | 60 | [link](https://github.com/cloudfoundry/cf-networking-release/blob/0c5029c88e9f61bb94829b4e0b8ed6732f30f9f0/jobs/policy-server-asg-syncer/spec#L31-L33) |
+| vxlan-policy-agent | asg_poll_interval_seconds | 60 | [link](https://github.com/cloudfoundry/silk-release/blob/f1606499925ff94bc036a641d688967c7af1fef4/jobs/vxlan-policy-agent/spec#L59-L61) |
+
+### 5. Did we stop passing ASG information through the LRP?
+No. ASGs are still passed through Cloud Controller to BBS to Rep to Garden to
+Silk CNI. This allows ASGs to be implemented correctly the minute an app is
+created.
```

Comment:
We call a force-asg-sync endpoint from the CNI, which overrides any of the data passed from garden to the CNI.

However, we still pass data through Diego to Garden for non-silk CNIs, so they keep their existing behavior.

Reply:
I decided not to edit the diagram to include the force-sync. I thought adding the container create flow made it too cluttered. I did add it in one of the FAQs though.